templates/search/search.html

```diff
@@ -216,7 +216,13 @@
         $("#filter_query").val(cleaned);
     }
 
-
+    function set_hidden_grouping_value(){
```

Comment:
Part of the reason for doing the original commit where this was removed was to get rid of javascript that isn't strictly required for some search features. In this case, restoring this function makes the checkbox work again, but after looking at the code I'm confused as to why we need it
https://github.com/MTG/freesound/blob/4e4b91fa45adabb5294487eeed384596425fd8bb/templates/search/advanced_search_form.html#L63-L64
Why don't we update it to make the checkbox the field named `g` instead of the hidden field? This will let us get rid of this extra js.
While looking at the change to `actual_grouping`, I also wonder why it's used. To me I don't see a reason to assign another variable and use it instead of `grouping`:
https://github.com/MTG/freesound/blob/e289acff01bf9159cb9c61d30fbfd8afef0c45f3/search/views.py#L162

Reply:
Yes you're right, I see no apparent reasons why these changes could not be done. I didn't do them before because I did not want to spend more time with the fix, but given that these are simple and easy to test (manually), I'll implement them.